wasmCloud/wasmCloud
# [FEAT] wash dev output improvements and fixes

## Feature or Problem
This PR, in order of commit:
- Rewords a couple help texts and reorders `eprintln` statements in wash dev to ensure more consistent and less duplicated output.
- Attempts to reorder some code logic to ensure the main devloop is where all output is printed from instead of being from nested functions, if possible.
- Ensures that wash dev is being run when only one host exists, allowing the existence of more than one host _only_ when a host ID is specified and that host is running. This also includes recommendations for users to run `wash down --all` to wipe to a clean slate in the case of bad state.
- Adds a spinner to the build process, during text output mode, to show that wash-cli is doing work and not just sitting there hoping for it to finish
- Fixes an issue where wadm wouldn't stop when up ctrl+c's
- Fixes an issue where wash up would exit with a failure finding the host PID
- Adds two integration tests for new functionality regarding multiple hosts

## Related Issues
Fixes https://github.com/wasmCloud/wasmCloud/issues/3609

## Release Information
wash-cli 0.37.1

## Consumer Impact


## Testing


### Unit Test(s)


### Acceptance or Integration


### Manual Verification
I did run through this and ensure that it all works as expected, but I plan to follow this draft up with one more commit to add an integration test for the new functionality. Here's an example!

https://github.com/user-attachments/assets/d31ff985-c39d-4646-a99c-17bda77e462d
